aws-logging-cloudwatch/src/androidTest/kotlin/com/amplifyframework/logging/cloudwatch/db/CloudWatchLoggingDatabaseInstrumentationTest.kt

```diff
@@ -0,0 +1,155 @@
+
+import org.junit.After
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import java.time.Instant
+
+@RunWith(AndroidJUnit4::class)
+@OptIn(ExperimentalCoroutinesApi::class)
+class CloudWatchLoggingDatabaseInstrumentationTest {
+    private val context = InstrumentationRegistry.getInstrumentation().context
+    private val testCoroutine = UnconfinedTestDispatcher()
+    private val loggingDbClass = CloudWatchLoggingDatabase(context, testCoroutine)
+    private val database by lazy {
+        System.loadLibrary("sqlcipher")
+        val passPhraseGetter = loggingDbClass.javaClass.getDeclaredMethod("getDatabasePassphrase", String::class.java)
+        passPhraseGetter.isAccessible = true
+        CloudWatchDatabaseHelper(context).getWritableDatabase(passPhraseGetter.invoke(loggingDbClass) as String)
+    }
+
+    private val testTimestamp1 = Instant.now().epochSecond
+    private val testTimestamp2 = Instant.now().epochSecond + 300
+    private val testTimestamp3 = Instant.now().epochSecond + 600
+    private val testTimestamp4 = Instant.now().epochSecond + 900
+    private val testCloudWatchLogEvent1 = CloudWatchLogEvent(testTimestamp1, "Customer Obsession")
+    private val testCloudWatchLogEvent2 = CloudWatchLogEvent(testTimestamp2, "Ownership")
+    private val testCloudWatchLogEvent3 = CloudWatchLogEvent(testTimestamp3, "Bias for Action")
+    private val testCloudWatchLogEvent4 = CloudWatchLogEvent(testTimestamp4, "Frugality")
+
+    @Before
+    fun setUp() = runTest {
+        Dispatchers.setMain(testCoroutine)
+    }
+
+    @After
+    fun tearDown() = runTest {
+        // Clear the db used for testing
+        loggingDbClass.clearDatabase()
+    }
+
+    /**
+     * This test verifies if the CloudWatch Logs are saved as intended by passing
+     * CloudWatchLogEvent to saveLogEvent() method
+     */
+    @Test
+    fun saveLogEvent_Saves_Two_Logs_After_Two_Calls() = runTest {
+        loggingDbClass.saveLogEvent(testCloudWatchLogEvent1)
+        loggingDbClass.saveLogEvent(testCloudWatchLogEvent2)
+
+        val savedLogs = loggingDbClass.queryAllEvents()
+
+        assertNotNull(savedLogs)
+        assertEquals(2, savedLogs.size)
+        assertEquals("Customer Obsession", savedLogs[0].message)
+        assertEquals(testTimestamp2, savedLogs[1].timestamp)
+        loggingDbClass.clearDatabase()
+    }
+
+    /**
+     * This test verifies if the CloudWatch Logs are retrieved
+     */
+    @Test
+    fun queryAllEvents_Successfully_Retrieves_Every_Log() = runTest {
+        assertEquals(0, loggingDbClass.queryAllEvents().size)
+        loggingDbClass.saveLogEvent(testCloudWatchLogEvent1)
+        assertEquals(1, loggingDbClass.queryAllEvents().size)
+        loggingDbClass.saveLogEvent(testCloudWatchLogEvent3)
+
+        val savedLogs = loggingDbClass.queryAllEvents()
+        assertEquals("Bias for Action", savedLogs[savedLogs.size -1].message)
+        loggingDbClass.clearDatabase()
```

Comment:
Can remove these `clearDatabase()` calls from each test as they are in the `teardown` function.

Reply:
Thank you for pointing out the redundancy here, went unnoticed. Removed!